smolyay/grid.py

```diff
@@ -2,348 +2,429 @@
 import itertools
 
 import numpy
+import scipy.stats.qmc
 
-from smolyay.basis import (BasisFunctionSet, NestedBasisFunctionSet)
+from smolyay.basis import BasisFunctionSet
 
 
-class IndexGridGenerator(abc.ABC):
-    """Grid points for making a surrogate.
+class MultidimensionalPointSet(abc.ABC):
+    """Multidimensional set of points
 
-    Depending on the approach and dimensionality,
-    a set of grid points, and their corresponding
-    basis functions are generated. These grid points
-    and functions can then be used for approximation
-    of complex systems.
+    Creates a set of grid points within the specified domain
 
     Parameters
     ----------
-    basis_set: :class:`BasisFunctionSet`
-        A set of unidimensional points and functions.
-
-    Raises
-    ------
-    TypeError
-        ``basis_set`` must be a :class:`BasisFunctionSet`.
+    domain: list
+         the domain of the grid
     """
 
-    def __init__(self, basis_set):
-        self.basis_set = basis_set
+    def __init__(self):
+        self._points = None
 
     @property
-    def basis_set(self):
-        """:class:`BasisFunctionSet`: basis set (points and functions)."""
-        return self._basis_set
-
-    @basis_set.setter
-    def basis_set(self, basis_set):
-        if not isinstance(basis_set, BasisFunctionSet):
-            raise TypeError('Basis set must be a BasisFunctionSet')
-        self._basis_set = basis_set
+    @abc.abstractmethod
+    def dimension(self):
```

Comment:
Should this be:
```suggestion
    def num_dimensions(self):
```
?

Also, this can be inferred from the shape of the domain, so it does not need to be abstract.
```
return self.domain.shape[0]
```

Reply:
It's always been referred to as "dimension" by the surrogate class and the grid generators.

Reply:
Yes, but we are **refactoring** this code, so things should be changed to be consistent. In this case, we prefix things with `num_` when they refer to a number of something.